rllib/algorithms/sac/sac.py

```diff
@@ -371,3 +435,149 @@ def get_default_policy_class(
             return SACTorchPolicy
         else:
             return SACTFPolicy
+
+    @override(DQN)
+    def training_step(self) -> ResultDict:
+        # If `RolloutWorker` is used, then use as the old stack `training step`
+        # of `DQN`.
+        if not self.config.uses_new_env_runners:
+            return super().training_step()
+
+        # Alternate between storing and sampling and training.
+        store_weight, sample_and_train_weight = calculate_rr_weights(self.config)
+        train_results = {}
+
+        # Run multiple sampling iterations.
+        for _ in range(store_weight):
+            # Otherwise, we use the new stack.
+            with self._timers[SAMPLE_TIMER]:
+                if self.workers.num_remote_workers() <= 0:
+                    episodes: List[SingleAgentEpisode] = [
+                        self.workers.local_worker().sample()
+                    ]
+                else:
+                    episodes: List[SingleAgentEpisode] = self.workers.foreach_worker(
+                        lambda w: w.sample(),
+                        local_worker=False,
+                    )
+                # Perform SAC postprocessing (Prio weights) on a (flattened)
+                # list of Episodes.
```

Comment:
erase this comment?

Reply:
DO we need to do any preprocessing here? Does a user pass in processing functions in the new stack anymore?